src/buttons/ButtonGroup.js

```diff
@@ -69,40 +74,76 @@ const ButtonGroup = props => {
-                borderBottomRightRadius: containerBorderRadius,
-              },
-              i === 0 && {
-                borderTopLeftRadius: containerBorderRadius,
-                borderBottomLeftRadius: containerBorderRadius,
+              !vertical &&
+                i < buttons.length - 1 && {
+                  borderRightWidth:
+                    i === 0
+                      ? 0
+                      : (innerBorderStyle && innerBorderStyle.width) || 1,
+                  borderRightColor:
+                    (innerBorderStyle && innerBorderStyle.color) ||
+                    colors.grey4,
+                },
+              !vertical &&
+                i === 1 && {
+                  borderLeftWidth:
+                    (innerBorderStyle && innerBorderStyle.width) || 1,
+                  borderLeftColor:
+                    (innerBorderStyle && innerBorderStyle.color) ||
+                    colors.grey4,
+                },
+              !vertical &&
+                i === buttons.length - 1 && {
+                  ...lastBorderStyle,
+                  borderTopRightRadius: containerBorderRadius,
+                  borderBottomRightRadius: containerBorderRadius,
+                },
+              !vertical &&
+                i === 0 && {
+                  borderTopLeftRadius: containerBorderRadius,
+                  borderBottomLeftRadius: containerBorderRadius,
+                },
+              vertical &&
+                i < buttons.length - 1 && {
+                  borderBottomWidth:
+                    i === 0
+                      ? 0
+                      : (innerBorderStyle && innerBorderStyle.width) || 1,
+                  borderBottomColor:
+                    (innerBorderStyle && innerBorderStyle.color) ||
+                    colors.grey4,
+                },
+              vertical &&
+                i === 1 && {
+                  borderTopWidth:
+                    (innerBorderStyle && innerBorderStyle.width) || 1,
+                  borderTopColor:
+                    (innerBorderStyle && innerBorderStyle.color) ||
+                    colors.grey4,
+                },
+              vertical &&
+                i === buttons.length - 1 && {
+                  ...lastBorderStyle,
+                  borderBottomLeftRadius: containerBorderRadius,
+                  borderBottomRightRadius: containerBorderRadius,
+                },
+              vertical &&
+                i === 0 && {
+                  borderTopLeftRadius: containerBorderRadius,
+                  borderTopRightRadius: containerBorderRadius,
+                },
+              isSelected && {
+                backgroundColor: colors.primary,
               },
             ]}
           >
             <View
               style={[
                 styles.textContainer,
                 buttonStyle && buttonStyle,
-                isSelected && {
-                  backgroundColor: colors.primary,
-                },
```

Comment:
I was talking about the other one lol, this one should stay here

Reply:
@xavier-villelegier did you mean the ones within the Component component? Cause when I deleted that one, I got this. 
<img width="107" alt="screen shot 2018-04-10 at 10 34 21 pm" src="https://user-images.githubusercontent.com/14284173/38593460-74a351d8-3d0f-11e8-81e0-8e923c1897fa.png">

Reply:
Hey @xavier-villelegier, I'm sorry I've been taking this long. I've been trying to figure out while preparing for a competition that is happening on the end of April. Would it be cool if I get back after the competition? I'll be trying to work on it as much as I can, but I'm kind of confused on how I can make the underlay cover the whole box.

Reply:
Something i tried was putting 
`vertical && isSelected && styles.verticalSelected,`
in the Component component. So that only when it's vertical, it takes it, But I'm assuming thats not the best way.

Reply:
@k3ithl1m Yea sure, take your time, competition first ! Good luck 💪 
Just do the simplest thing you can think of and we'll review it, it's not critical so we can take the time to do it properly.

Reply:
Hey @xavier-villelegier sorry I got back to this late. I had finals last week too. But I've been trying a few ways and most of it doesnt work unless i put
`isSelected && vertical && {
-                  backgroundColor: colors.primary,
-                },`

So I wasn't sure whether that was right too.